plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/OpenShiftImages.java

```diff
@@ -30,4 +33,21 @@
 	public static final Image GEAR_IMG = repo.getImage("gear.png"); //$NON-NLS-1$ 
 	public static final ImageDescriptor LAYER = repo.create("layer.png"); //$NON-NLS-1$ 
 	public static final Image LAYER_IMG = repo.getImage("layer.png"); //$NON-NLS-1$ 
+	
+	public static Image getImage(IResource resource) {
```

Comment:
@adietish reverting this change for now as its only used it in one place, but I don't understand your point really.  The images are static entities so adding a method here is just returning a static entity.  I don't see in our label providers where they are explicitly disposed.  My original thought was to reuse the switch statement instead of repeating, or introducing a utility class which does exactly this.